### ISSUE TITLE
Accessibility: Page title and H1 don't accurately reflect the page content in the My Account sub pages 

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43632]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

As users navigate through the My Account subpages (Dashboard, Orders, Addresses, Payment Methods, Account Details, Edit Address, etc.), the title of the page and the H1 on the page never changes, though the URL does. It always says "My Account" and never accurately reflects which subpage users are viewing.

Web Content Accessibility Guidelines (WCAG) requires that that [pages have accurate titles](https://www.w3.org/WAI/WCAG21/Understanding/page-titled.html) and [fails pages if the title doesn't identify the contents.](https://www.w3.org/WAI/WCAG21/Techniques/failures/F25) It also required that [headings accurately describe the content.](https://www.w3.org/WAI/WCAG22/Understanding/headings-and-labels.html)

### Expected behavior

When navigating through the My Account pages, as the URL changes, the page title and H1 should change to reflect the contents of the sub-page that the user is viewing.

### Actual behavior

Described above.

### Steps to reproduce

Click through the sub pages on the my account page and observe that the H1 and page title (visible in the browser tab) never change.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by amesplant** [comment]
It would appear that this is where the title should be getting updated:
[plugins/woocommerce/includes/wc-page-functions.php#L19](https://github.com/woocommerce/woocommerce/blob/62164085114b9a1fe2f22f55f4874780f7cacae0/plugins/woocommerce/includes/wc-page-functions.php#L19)

But on the `in_the_looop()` is firing as `false` and even if that is removed, the title is not being updated on the page.

**Comment by amesplant** [comment]
Note from @Sidsector9 

This is because the document title for My Account page is set by the value returned by `single_post_title()`. We'll have to set the title by using the `pre_get_document_title` filter hook.

**Comment by Manussakis** [comment]
Thanks @amesplant for the insights on how to fix this issue. I am taking it over now.
